File: SplashScreenApp/app/src/main/java/kevin/codelab/splashscreenapp/MainActivity.java
package kevin.codelab.splashscreenapp;

import androidx.annotation.Nullable;
import androidx.annotation.RequiresApi;
import androidx.appcompat.app.AppCompatActivity;

import android.animation.Animator;
import android.animation.AnimatorListenerAdapter;
import android.animation.ObjectAnimator;
import android.os.Build;
import android.os.Bundle;
import android.view.View;
import android.view.animation.AnticipateInterpolator;

import kevin.codelab.splashscreenapp.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    ActivityMainBinding binding;

    @RequiresApi(api = Build.VERSION_CODES.S)
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        getSplashScreen().setOnExitAnimationListener(splashScreenView -> {
            final ObjectAnimator slideUp = ObjectAnimator.ofFloat(
                    splashScreenView,
                    View.TRANSLATION_X,
                    0f,
                    -splashScreenView.getHeight()
            );
            slideUp.setInterpolator(new AnticipateInterpolator());
            slideUp.setDuration(200L);

            // Call SplashScreenView.remove at the end of your custom animation.
            slideUp.addListener(new AnimatorListenerAdapter() {
                @Override
                public void onAnimationEnd(Animator animation) {
                    splashScreenView.remove();
                }
            });

            // Run your animation.
            slideUp.start();
        });
        setContentView(binding.getRoot());


        // final View content = findViewById(R.id.content);
//        content.getViewTreeObserver().addOnPreDrawListener(
//                new ViewTreeObserver.OnPreDrawListener() {
//                    @Override
//                    public boolean onPreDraw() {
//
//                        if (mViewModel.isReady()) {
//                            content.getViewTreeObserver().removeOnPreDrawListener(this);
//                            return true;
//                        } else {
//                            return false;
//                        }
//
//                    }
//                }
//        );
    }


}

File: SplashScreen/app/src/main/java/kevin/codelab/splashscreen/MainActivity.kt
package kevin.codelab.splashscreen

import android.animation.ObjectAnimator
import android.os.Build
import android.os.Bundle
import android.view.View
import android.view.animation.AnticipateInterpolator
import androidx.annotation.RequiresApi
import androidx.appcompat.app.AppCompatActivity
import androidx.core.animation.doOnEnd
import kevin.codelab.splashscreen.databinding.ActivityMainBinding
import java.time.Instant
import javax.xml.datatype.Duration

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    @RequiresApi(Build.VERSION_CODES.S)
    override fun onCreate(savedInstanceState: Bundle?) {
        binding = ActivityMainBinding.inflate(layoutInflater)
        super.onCreate(savedInstanceState)

        splashScreen.setOnExitAnimationListener { splashScreenView ->
            val slideUp = ObjectAnimator.ofFloat(
                splashScreenView,
                View.TRANSLATION_Y,
                0f,
                -splashScreenView.height.toFloat()
            )
            slideUp.interpolator = AnticipateInterpolator()
            slideUp.duration = 200L

            slideUp.doOnEnd {
                splashScreenView.remove()
            }

            slideUp.start()

            // Get the duration of the animated vector drawable.
            val animationDuration = splashScreenView.iconAnimationDuration
// Get the start time of the animation.
            val animationStart = splashScreenView.iconAnimationStart
// Calculate the remaining duration of the animation.
            val remainingDuration = if (animationDuration != null && animationStart != null) {
                (animationDuration - Duration.between(animationStart, Instant.now()))
                    .toMillis()
                    .coerceAtLeast(0L)
            } else {
                0L
            }

            setContentView(binding.root)
        }
    }


}